datastore/errors/errors_badger.go

```diff
@@ -0,0 +1,21 @@
+// Copyright 2024 Democratized Data Foundation
+//
+// Use of this software is governed by the Business Source License
+// included in the file licenses/BSL.txt.
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0, included in the file
+// licenses/APL.txt.
+
+//go:build !js
+
+package errors
+
+import "github.com/sourcenetwork/badger/v4"
+
+func init() {
+	ErrTxnConflict = badger.ErrConflict
```

Comment:
suggestion: I can picture this causing problems - not building for JS does not mean that the errors should be badger ones!  Other stores may be used (also or as well).  It would be far better IMO to make sure that we return consistent errors instead of doing this.

In my opinion errors are part of the interface (contract), not the implementation - yes, a bunch of annoying libs break this, and occasionally it is necessary if the interface does not declare enough useful errors (not a problem here, as we have control - even if it means a thin wrapper), but this is the point of view I have when making the above suggestion.

Reply:
I've added a function to convert the errors within the badger datastore. It was much simpler than I first thought.